### PR TITLE
[AND-3385] Open Bidding implementation

### PR DESCRIPTION
**Pull request by HuFangshuai**
PR for ticket: https://vungle.atlassian.net/browse/AND-3385
Document: https://docs.google.com/document/d/1e5IOlr2tjN-WKsqH1hwzvKWSK8oUiUL96zzaxB4SFpg/edit?usp=sharing

TODO: testing

### CODE REVIEW

**Comment by jeffreychang-vungle** [comment]
Are there any logic changes beyond just updated the base classes? Lots of code change

**Comment by HuFangshuai** [comment]
> Are there any logic changes beyond just updated the base classes? Lots of code change

Yeah, there's another branch feature-AND-3385-update for simple implementation, but without adMarkUp logic. Because I am not sure the `adMarkUp` field whether can be passed by `serverExtra` parameter. 
In this PR, AdMob provided new API and `getBidResponse()` method, that's why I refactored lots of code here.
When we are ready for testing, I will check the `serverExtra` again to make sure if we can make simple changes for this feature.

**Comment by jeffchang5** [comment]
@HuFangshuai is this ready to be reviewed?

**Comment by HuFangshuai** [comment]
> @HuFangshuai is this ready to be reviewed?

AdMob hasn't completed the integration of this feature. We can review it later.

**Comment by jiancchen** [comment]
@HuFangshuai conflicts w/ 6.10.x needs update

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleMediationAdapter.java [hunk 1]
```diff
@@ -371,4 +395,22 @@ public String getType() {
       return mType;
     }
   }
+
+  @Override
+  public void loadInterstitialAd(
```
Comment (jiancchen) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleMediationAdapter.java:400]: minor formatting on both functions

**Comment by HuFangshuai** [comment]
> looks good. we might want to merge changes from upstream master first to avoid seeing 89 commits/68 files changed and have Vungle-specific metrics listed only :)

Thanks, @anton-blashchuk-zoolatech. Updated. There are 10 commits and 7 files changed.

**Pull request merged**